SimplenoteUITests/SimplenoteUITestsNoteEditor.swift

```diff
@@ -312,4 +312,49 @@ class SimplenoteUISmokeTestsNoteEditor: XCTestCase {
         TrashAssert.noteExists(noteName: noteTitle + "-1")
         TrashAssert.noteExists(noteName: noteTitle + "-2")
     }
+
+    func testUndoTrashFromSnackbar() throws {
+        trackTest()
+        let noteName = "Note Title"
+
+        trackStep()
+        populateNoteList(title: noteName, numberToCreate: 1)
+
+        trackStep()
+        NoteList.longPressNote(title: noteName + "-1")
+        NoteList.deleteNoteFromContextMenu()
+        NoteListAssert.notesNumber(expectedNotesNumber: 0)
+
+        trackStep()
+        NoteList.tapUndoOnSnackbar()
+        NoteListAssert.notesNumber(expectedNotesNumber: 1)
+        NoteListAssert.noteExists(noteName: noteName + "-1")
+    }
+
+    func testCopyAndPasteInterlink() throws {
+        trackTest()
+        let toCopyNoteTitle = "Copy Internal Link Title"
+        let toPasteNoteTitle = "Paste Internal Link Title\n"
+        let note = NoteData(
+            name: "Paste Internal Link Title",
+            content: "[Copy Internal Link Title-1](simplenote://note/"
+        )
```

Comment:
There are several ways to improve this situation a bit:

1. Keeping in mind that you can have a line break automatically added by `NoteEditor.enterTitle`, you no longer need to have "\n" at the end of `toPasteNoteTitle`
2. Since you create only one note via `populateNoteList`, you can use `NoteList.createNoteAndLeaveEditor` instead, and then you have no need to have "-1" in NoteData content.
3. You could adjust the naming a bit, so that the names reflect clearer what are they for.
4. Use [string interpolation](https://www.hackingwithswift.com/read/0/5/string-interpolation), since now you can refer both strings in the note data

In the end, it can look something like this:

```suggestion
        let originalNoteTitle = "Original Note Title"
        let referringNoteTitle = "Referring Note Title"
        let referringNoteData = NoteData(
            name: referringNoteTitle
            content: "[\(originalNoteTitle)](simplenote://note/"
        )
```

Reply:
Thanks for the link on string interpolation, I was actually looking for something similar to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) on swift, now I know what it is 😄